plugin/src/main/java/com/spotify/plugin/dockerfile/AbstractDockerMojo.java

```diff
@@ -449,6 +454,8 @@ private RegistryAuthSupplier createRegistryAuthSupplier() {
       getLog().info("Google Container Registry support is disabled");
     }
 
+    suppliers.add(new MavenPomAuthSupplier(this));
```

Comment:
how about inspecting the username here to make sure it is set before adding it to the suppliers list rather than having to have those checks in MavenPomAuthSupplier?

Reply:
👍